kolibri/core/assets/src/conditionalPromise.js

```diff
@@ -14,12 +14,12 @@ export default class ConditionalPromise {
   }
 
   catch(...args) {
-    this._promise.catch(...args);
+    this._promise = this._promise.catch(...args);
```

Comment:
commit 57ac9b7f4f40215d4cfd61ba5d4ce84fd6c37055 should not be part of this PR.

to make a new PR with only this work, please:

* make a new branch
* `reset --hard` to the 0.8.x branch
* cherry-pick the three commits in this branch
* submit a new PR

thanks!